tests/lapi/debug_getinfo_test.lua

```diff
@@ -0,0 +1,92 @@
+--[[
+SPDX-License-Identifier: ISC
+Copyright (c) 2023-2025, Sergey Bronnikov.
+
+Parameter 'what' of 'debug.getinfo' cannot start with '>',
+https://www.lua.org/bugs.html#5.4.2-2
+
+Access to debug information in line hook of stripped function,
+https://github.com/lua/lua/commit/ae5b5ba529753c7a653901ffc29b5ea24c3fdf3a
+
+Return hook may not see correct values for active local variables when function returns,
+https://www.lua.org/bugs.html#5.3.0-4
+
+The PC out-of-range in lj_debug_frameline(),
+https://github.com/LuaJIT/LuaJIT/issues/1369
+
+LuaJIT segfault in debug.getinfo(),
+https://github.com/LuaJIT/LuaJIT/issues/509
+
+Synopsis: debug.getinfo([thread,] function [, what])
+]]
+
+local luzer = require("luzer")
+local test_lib = require("lib")
+
+local what = {
+    "n", -- fills in the field name and namewhat.
+    "S", -- fills in the fields source, short_src, linedefined,
+         -- lastlinedefined, and what.
+    "l", -- fills in the field currentline.
+    "u", -- fills in the field nups.
+    "f", -- pushes onto the stack the function that is running at
+         -- the given level.
+    "L", -- pushes onto the stack a table whose indices are the
+         -- numbers of the lines that are valid on the function.
+}
+
+local what_modes
+
+local hook_mask = {
+    "c", -- the hook is called every time Lua calls a function.
+    "r", -- the hook is called every time Lua returns from a
+         -- function.
+    "l", -- the hook is called every time Lua enters a new line of
+         -- code.
+}
+
+local loadstring = type(loadstring) == "function" and loadstring or load
+
+local function debug_hook()
+    debug.getinfo(1, table.concat(what_modes))
```

Comment:
Maybe it is more interestring to do in cycle something like the following:
```lua
local lvl = 0
local modes = table.concat(what_modes)
while debug.getinfo(lvl, modes) do
        lvl = lvl + 1
end
```

Reply:
Updated.